plugins/browser-plugin-vimeo-tracking/src/mediaPluginBinding.ts

```diff
@@ -0,0 +1,215 @@
+import { Error as VimeoError, QualityChangeEvent, VimeoCuePointData } from '@vimeo/player';
+
+import {
+  trackMediaBufferEnd,
+  trackMediaBufferStart,
+  trackMediaEnd,
+  trackMediaFullscreenChange,
+  trackMediaPause,
+  trackMediaPictureInPictureChange,
+  trackMediaPlay,
+  trackMediaPlaybackRateChange,
+  trackMediaQualityChange,
+  trackMediaReady,
+  trackMediaSeekEnd,
+  trackMediaSeekStart,
+  trackMediaSelfDescribingEvent,
+  trackMediaVolumeChange,
+  updateMediaTracking,
+} from '@snowplow/browser-plugin-media';
+import { MediaEventType } from '@snowplow/browser-plugin-media/src/types';
+
+import { getVimeoMetadata, getVimeoMediaAttributes } from './utils';
+import { VimeoSchema } from './schema';
+import { InternalVimeoEvent } from './events';
+import {
+  TrackedPlayer,
+  ChapterData,
+  EventData,
+  TextTrackData,
+  TimeUpdateData,
+  VimeoInteractiveHotspotClickedData,
+  VimeoPlayerCuePointData,
+  EventType,
+} from './types';
+
+export function trackEvent(
+  trackedPlayer: TrackedPlayer,
+  eventType: EventType,
+  eventData: EventData | unknown = {},
+  trackError: (error: VimeoError) => void
+): void {
+  getVimeoMediaAttributes(trackedPlayer)
+    .then((player) => {
+      updateMediaTracking({
+        id: trackedPlayer.config.id,
+        player,
```

Comment:
Nit: this is fine to separate the `update` and `track` calls but wouldn't it be better to pass the `player` property directly in the `track` calls and avoid this extra update one?